### PR TITLE
Update Bazel rules for Go to v0.24.7

### PR DESCRIPTION
**Pull request by hakman**
/cc @justinsb @mikesplain @rifelpet

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -169,15 +102,29 @@ go_repository(
...
+
+container_pull(
+    name = "distroless_base_arm64",
+    digest = "sha256:c60be29941a0be6f748c8cf2e42832f95e9b73276042d3c44212af7cf4a152c9",
+    registry = "gcr.io",
+    repository = "distroless/base",
+)
+
+container_pull(
+    name = "distroless_base_debug_amd64",
```
Comment (rifelpet) [inline WORKSPACE:119]: where are these debug images used?

Comment (hakman) [reply WORKSPACE:119]: These are not being used, but are there for easier debugging during development.
They were there before, so this is the main reason for keeping them.

File: Makefile [hunk 2]
```diff
@@ -564,48 +564,44 @@ bazel-test:
...
-	bazel ${BAZEL_OPTIONS} build ${BAZEL_CONFIG} --features=pure --platforms=@io_bazel_rules_go//go/toolchain:linux_arm64 //protokube/...
+	bazel ${BAZEL_OPTIONS} build ${BAZEL_CONFIG} --@io_bazel_rules_go//go/config:pure --platforms=@io_bazel_rules_go//go/toolchain:linux_amd64 //protokube/...
+	bazel ${BAZEL_OPTIONS} build ${BAZEL_CONFIG} --@io_bazel_rules_go//go/config:pure --platforms=@io_bazel_rules_go//go/toolchain:linux_arm64 //protokube/...
 
 .PHONY: bazel-crossbuild-protokube-image
 bazel-crossbuild-protokube-image:
 	bazel ${BAZEL_OPTIONS} build ${BAZEL_CONFIG} --platforms=@io_bazel_rules_go//go/toolchain:linux_amd64 //protokube/cmd/protokube:image-bundle-amd64.tar
 	bazel ${BAZEL_OPTIONS} build ${BAZEL_CONFIG} --platforms=@io_bazel_rules_go//go/toolchain:linux_arm64 //protokube/cmd/protokube:image-bundle-arm64.tar
 
-.PHONY: bazel-crossbuild-node-authorizer-image
```
Comment (rifelpet) [inline Makefile:599]: why is this being removed? Did we formalize a deprecation plan for node-authorizer? I can't find anything in the release notes

Comment (hakman) [reply Makefile:599]: The bazel target is unmaintained, only the Docker one is working.

**Comment by rifelpet** [comment]
glad we're able to get over the 0.23 wall :)

**Comment by hakman** [comment]
> glad we're able to get over the 0.23 wall :)

And without some ugly hack...

**Pull request merged**